### PR TITLE
setup webdriver with a browser and waits

### PR DESCRIPTION
**Pull request by raulchoque-eng**
Add files to create a browser with waits read data of a properties file

### CODE REVIEW

**Review by MelissaRoman06 — APPROVED** [review]

File: .gitignore [hunk 1]
```diff
@@ -46,3 +46,7 @@ hs_err_pid*
 
 #Log4J
 /log/*
+
+# ignore this folder, except java files
+src/main/java/core/selenium/webdrivers/driver
+config/
```
Comment (titor-oopart) [inline .gitignore]: add empty line

File: src/main/java/core/StrategyToCreate.java [hunk 2]
```diff
@@ -0,0 +1,20 @@
+package core;
```
Comment (titor-oopart) [inline src/main/java/core/StrategyToCreate.java:13]: add Copyright

File: src/main/java/core/StrategyToCreate.java [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+
+import core.selenium.webdrivers.Browser;
+
+/**
+ * StrategyCreate class.
```
Comment (titor-oopart) [inline src/main/java/core/StrategyToCreate.java]: the class name is incorrect

File: src/main/java/core/selenium/ProjectProperties.java [hunk 4]
```diff
@@ -0,0 +1,37 @@
+package core.selenium;
```
Comment (titor-oopart) [inline src/main/java/core/selenium/ProjectProperties.java:13]: add documentation Copyright

File: src/main/java/core/selenium/ProjectProperties.java [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+     * Constructor of ProjectProperties enum.
+     *
+     * @param value value for create a ProjectProperties enum.
+     */
+    ProjectProperties(final String value) {
+        this.value = value;
+    }
+
+    /**
+     * gets value of value constant of this enum.
```
Comment (titor-oopart) [inline src/main/java/core/selenium/ProjectProperties.java]: the documentation needs to starts with capital letters

File: src/main/java/core/selenium/WebDriverConfig.java [hunk 6]
```diff
@@ -0,0 +1,71 @@
...
+    private int implicitWait;
+    private int explicitWait;
+
+    private static WebDriverConfig webDriverConfig;
+
+    /**
+     * Constructor of WebDriverConfig class.
+     */
+    private WebDriverConfig() {
+
```
Comment (titor-oopart) [inline src/main/java/core/selenium/WebDriverConfig.java]: delete this line

File: src/main/java/core/selenium/WebDriverConfig.java [hunk 7]
```diff
@@ -0,0 +1,71 @@
...
+    /**
+     * Gets nameBrowser attribute of this class.
+     *
+     * @return as String the nameBrowser attribute of this class.
+     */
+    public String getNameBrowser() {
+        return nameBrowser;
+    }
+
+
```
Comment (titor-oopart) [inline src/main/java/core/selenium/WebDriverConfig.java]: delete unnecessary line

File: src/main/java/core/selenium/webdrivers/Chrome.java [hunk 8]
```diff
@@ -0,0 +1,48 @@
...
+        System.setProperty("webdriver.chrome.driver", PathWebDriver.PATH_CHROME_DRIVER.getPathDriver());
+        this.driver = new ChromeDriver();
+    }
+
+    /**
+     * Gets single instance of Chrome class.
+     *
+     * @return a Chrome object.
+     */
+    public static Chrome getInstance() {
```
Comment (titor-oopart) [inline src/main/java/core/selenium/webdrivers/Chrome.java]: This class just needs the method for create driver singleton is unnecessary

File: src/main/java/core/selenium/webdrivers/FactoryBrowser.java [hunk 9]
```diff
@@ -0,0 +1,38 @@
+package core.selenium.webdrivers;
```
Comment (titor-oopart) [inline src/main/java/core/selenium/webdrivers/FactoryBrowser.java:13]: put documentation copyright

File: src/main/java/core/selenium/webdrivers/Firefox.java [hunk 10]
```diff
@@ -0,0 +1,47 @@
...
+        System.setProperty("webdriver.gecko.driver", PathWebDriver.PATH_GECKO_DRIVER.getPathDriver());
+        this.driver = new FirefoxDriver();
+    }
+
+    /**
+     * Gets single instance of Firefox class.
+     *
+     * @return a Firefox object.
+     */
+    public static Firefox getInstance() {
```
Comment (titor-oopart) [inline src/main/java/core/selenium/webdrivers/Firefox.java]: unnecessary singleton this class can be more simple

File: src/main/java/core/selenium/webdrivers/PathWebDriver.java [hunk 11]
```diff
@@ -0,0 +1,34 @@
+package core.selenium.webdrivers;
```
Comment (titor-oopart) [inline src/main/java/core/selenium/webdrivers/PathWebDriver.java:13]: add documentation copyright

File: src/main/java/core/utils/ReadFile.java [hunk 12]
```diff
@@ -0,0 +1,33 @@
...
+    public static Properties getInstance() {
+        Properties prop = new Properties();
+        try (InputStream input = new FileInputStream("gradle.properties")) {
+            prop.load(input);
+        } catch (FileNotFoundException ex) {
+            throw new NullPointerException("This file not exist!!!");
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+        return prop;
```
Comment (titor-oopart) [inline src/main/java/core/utils/ReadFile.java]: I think that prop can be a better name

File: src/main/java/core/utils/ReadFile.java [hunk 13]
```diff
@@ -0,0 +1,33 @@
+package core.utils;
```
Comment (titor-oopart) [inline src/main/java/core/utils/ReadFile.java:13]: add copyright

File: src/main/java/core/selenium/webdrivers/Firefox.java [hunk 14]
```diff
@@ -0,0 +1,47 @@
+package core.selenium.webdrivers;
```
Comment (titor-oopart) [inline src/main/java/core/selenium/webdrivers/Firefox.java:13]: add copyright

File: src/main/java/core/selenium/webdrivers/Browser.java [hunk 15]
```diff
@@ -0,0 +1,20 @@
+package core.selenium.webdrivers;
```
Comment (titor-oopart) [inline src/main/java/core/selenium/webdrivers/Browser.java:13]: add copyright

File: src/main/java/core/selenium/WebDriverManager.java [hunk 16]
```diff
@@ -0,0 +1,75 @@
+package core.selenium;
```
Comment (titor-oopart) [inline src/main/java/core/selenium/WebDriverManager.java:13]: add documentation

File: src/main/java/core/selenium/WebDriverConfig.java [hunk 17]
```diff
@@ -0,0 +1,71 @@
+package core.selenium;
```
Comment (titor-oopart) [inline src/main/java/core/selenium/WebDriverConfig.java:13]: add copyright

File: .gitignore (hunk 1 above)
Comment (raulchoque-eng) [reply .gitignore]: Thank you, this wrong was fixed.

File: src/main/java/core/StrategyToCreate.java (hunk 2 above)
Comment (raulchoque-eng) [reply src/main/java/core/StrategyToCreate.java:13]: Thank you, this wrong was fixed.

File: src/main/java/core/StrategyToCreate.java (hunk 3 above)
Comment (raulchoque-eng) [reply src/main/java/core/StrategyToCreate.java]: Thank you, this wrong was fixed.

File: src/main/java/core/selenium/ProjectProperties.java (hunk 4 above)
Comment (raulchoque-eng) [reply src/main/java/core/selenium/ProjectProperties.java:13]: Thank you, this wrong was fixed.

File: src/main/java/core/selenium/ProjectProperties.java (hunk 5 above)
Comment (raulchoque-eng) [reply src/main/java/core/selenium/ProjectProperties.java]: Thank you, this wrong was fixed.

File: src/main/java/core/utils/ReadFile.java (hunk 13 above)
Comment (raulchoque-eng) [reply src/main/java/core/utils/ReadFile.java:13]: Thank you, this wrong was fixed.

File: src/main/java/core/selenium/webdrivers/Firefox.java (hunk 14 above)
Comment (raulchoque-eng) [reply src/main/java/core/selenium/webdrivers/Firefox.java:13]: Thank you, this wrong was fixed.

File: src/main/java/core/selenium/webdrivers/Browser.java (hunk 15 above)
Comment (raulchoque-eng) [reply src/main/java/core/selenium/webdrivers/Browser.java:13]: Thank you, this wrong was fixed.

File: src/main/java/core/selenium/WebDriverManager.java (hunk 16 above)
Comment (raulchoque-eng) [reply src/main/java/core/selenium/WebDriverManager.java:13]: Thank you, this wrong was fixed.

File: src/main/java/core/selenium/WebDriverConfig.java (hunk 17 above)
Comment (raulchoque-eng) [reply src/main/java/core/selenium/WebDriverConfig.java:13]: Thank you, this wrong was fixed.

File: src/main/java/core/selenium/WebDriverConfig.java (hunk 6 above)
Comment (raulchoque-eng) [reply src/main/java/core/selenium/WebDriverConfig.java]: Thank you, this wrong was implement.

File: src/main/java/core/selenium/WebDriverConfig.java (hunk 7 above)
Comment (raulchoque-eng) [reply src/main/java/core/selenium/WebDriverConfig.java]: Thank you, this wrong was fixed.

File: src/main/java/core/selenium/webdrivers/FactoryBrowser.java (hunk 9 above)
Comment (raulchoque-eng) [reply src/main/java/core/selenium/webdrivers/FactoryBrowser.java:13]: Thank you, this wrong was fixed.

File: src/main/java/core/selenium/webdrivers/PathWebDriver.java (hunk 11 above)
Comment (raulchoque-eng) [reply src/main/java/core/selenium/webdrivers/PathWebDriver.java:13]: Thank you, this wrong was fixed.

File: src/main/java/core/utils/ReadFile.java (hunk 12 above)
Comment (raulchoque-eng) [reply src/main/java/core/utils/ReadFile.java]: Thank you, this wrong was fixed.

File: src/main/java/core/selenium/webdrivers/Firefox.java (hunk 10 above)
Comment (raulchoque-eng) [reply src/main/java/core/selenium/webdrivers/Firefox.java]: Thank you, this wrong was fixed.

File: src/main/java/core/selenium/webdrivers/Chrome.java (hunk 8 above)
Comment (raulchoque-eng) [reply src/main/java/core/selenium/webdrivers/Chrome.java]: Thank you, this wrong was fixed.

**Review by titor-oopart — APPROVED** [review]

**Pull request merged**